spi/transfer-spi/src/main/java/org/eclipse/dataspaceconnector/spi/transfer/provision/ProvisionResult.java

```diff
@@ -0,0 +1,39 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.spi.transfer.provision;
+
+import org.eclipse.dataspaceconnector.spi.response.ResponseFailure;
+import org.eclipse.dataspaceconnector.spi.response.ResponseStatus;
+import org.eclipse.dataspaceconnector.spi.result.AbstractResult;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.ProvisionResponse;
+
+import java.util.List;
+
+/**
+ * The result of a provision operation.
+ */
+public class ProvisionResult extends AbstractResult<ProvisionResponse, ResponseFailure> {
```

Comment:
This result is pretty much the same as:
`DeprovisionResult`, `TransferInitiateResult`, `DataFlowInitiateResult`, `TransferResult`.
Could be defined as:
`public class StatusResult<T> extends AbstractResult<T, ResponseFailure>`
that is a result that, in case of failure gives a `ResponseFailure`, that contains a `Failure`.
Probably we could address this refactor to another issue.

Reply:
+1